src/chronos/wTraveler/update-user-handler.ts

```diff
@@ -15,7 +16,38 @@ export function handleUpdateUser(event: UpdateUser): void {
   const user = event.params.user.toHexString();
   const expires = event.params.expires;
 
+  if (expires.lt(event.block.timestamp)) {
+    log.warning(
+      "[handleUpdateUser] Expiration date is in the past, ignoring event, tx: {}",
+      [event.transaction.hash.toHex()]
+    );
+    return;
+  }
+
   const nft = loadNft(TYPE, tokenId);
+  const currentRentalId = nft.currentRental;
+
+  if (user == ZERO_ADDRESS) {
+    if (!currentRentalId) {
+      log.warning(
+        "[handleUpdateUser] No current rental for NFT: {}, user: {}, tx: {}",
+        [nft.id, user, event.transaction.hash.toHex()]
+      );
+      return;
+    }
+
+    const currentRental = loadRental(currentRentalId!);
+    currentRental.expiration_date = event.block.timestamp;
+    currentRental.save();
```

Comment:
You must also check if the rental is expired before setting the `expiration_date`as `event.block.timestamp`. If the rental is already expired, there is no point on updating it.